src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -280,6 +280,70 @@ public TimeSpan Expect100ContinueTimeout
             }
         }
 
+        /// <summary>
+        /// Gets or sets the keep alive ping delay. The client will send a keep alive ping to the server if it
+        /// doesn't receive any frames on a connection for this period of time. This property is used together with
+        /// <see cref="SocketsHttpHandler.KeepAlivePingTimeout"/> to close broken connections.
+        /// <para>
+        /// Delay value must be greater than or equal to 1 second. Set to <see cref="Timeout.InfiniteTimeSpan"/> to
+        /// disable the keep alive ping.
+        /// Defaults to <see cref="Timeout.InfiniteTimeSpan"/>.
+        /// </para>
+        /// </summary>
+        public TimeSpan KeepAlivePingDelay
+        {
+            get => _settings._keepAlivePingDelay;
+            set
+            {
+                if (value.Ticks < TimeSpan.TicksPerSecond && value != Timeout.InfiniteTimeSpan)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), value, SR.Format(SR.net_http_value_must_be_greater_than_or_equal, "1 second"));
```

Comment:
Although we dont localize our exception messages today, we may in future, and when we do that ideally we won't need code changes. Unfortunately if we create messages by concatenating sentence fragments it is not localizable. Eg., consider if we must localize such that "second" has to grammatically agree with the rest of the sentence, or "1" and "second" must not be next to each other. Possible approaches
1. "The specified value must be greater than or equal to {0} seconds." 
1. "The specified value must be greater than or equal to &quot;{0}&quot;"

I think #1 is what we usually do.

Reply:
Fixed.